tools/generate_schema_wrapper.py

```diff
@@ -861,59 +865,65 @@ def vegalite_main(skip_download: bool = False) -> None:
+
     for channel, info in channel_infos.items():
-        field_class_name = info.field_class_name
-        assert (
-            field_class_name is not None
-        ), "All channels are expected to have a field class"
-        datum_and_value_class_names = []
-        if info.datum_class_name is not None:
-            datum_and_value_class_names.append(info.datum_class_name)
-
-        if info.value_class_name is not None:
-            datum_and_value_class_names.append(info.value_class_name)
-
-        # dict stands for the return types of alt.datum, alt.value as well as
-        # the dictionary representation of an encoding channel class. See
-        # discussions in https://github.com/vega/altair/pull/3208
-        # for more background.
-        union_types = ["str", field_class_name, "Map"]
-        docstring_union_types = ["str", rst_syntax_for_class(field_class_name), "Dict"]
+        alias_name: str = f"Channel{channel[0].upper()}{channel[1:]}"
+
+        it: Iterator[str] = info.all_names
+        it_rst_names: Iterator[str] = (rst_syntax_for_class(c) for c in info.all_names)
+
+        docstring_types: list[str] = ["str", next(it_rst_names), "Dict"]
+        tp_inner: str = ", ".join(chain(("str", next(it), "Map"), it))
+        tp_inner = f"Union[{tp_inner}]"
+
         if info.supports_arrays:
-            # We could be more specific about what types are accepted in the list
-            # but then the signatures would get rather long and less useful
-            # to a user when it shows up in their IDE.
-            union_types.append("list")
-            docstring_union_types.append("List")
-
-        union_types = union_types + datum_and_value_class_names
-        docstring_union_types = docstring_union_types + [
-            rst_syntax_for_class(c) for c in datum_and_value_class_names
-        ]
+            docstring_types.append("List")
+            tp_inner = f"OneOrSeq[{tp_inner}]"
 
-        signature_args.append(
-            f"{channel}: Optional[Union[{', '.join(union_types)}]] = Undefined"
-        )
+        doc_types_flat: str = ", ".join(chain(docstring_types, it_rst_names))
 
-        docstring_parameters.extend(
-            (
-                f"{channel} : {', '.join(docstring_union_types)}",
-                f"    {process_description(info.deep_description)}",
-            )
-        )
-    if len(docstring_parameters) > 1:
-        docstring_parameters += [""]
-    docstring = indent_docstring(
-        docstring_parameters, indent_level=8, width=100, lstrip=False
+        type_aliases.append(f"{alias_name}: TypeAlias = {tp_inner}")
+        typed_dict_args.append(f"{channel}: {alias_name}")
+        signature_args.append(f"{channel}: Optional[{alias_name}] = Undefined")
+
+        description: str = f"    {process_description(info.deep_description)}"
+
+        signature_doc_params.extend((f"{channel} : {doc_types_flat}", description))
+        typed_dict_doc_params.extend((f"{channel}", description))
```

Comment:
@binste I had to lower the resolution to get below 10MB, but hopefully you can still see what the issue is.


https://github.com/user-attachments/assets/9f72ba05-e376-48cf-8182-672122483927

Essentially, the `TypeAlias`'s make it more difficult to find out what the correct type is.
I still think we need them for the individual args in `polars`, but I think they only worsen the UX for the `TypedDict` case - since you need to dig through refs to find the right types.

Simply using the expanded types still keeps everything on one row, but lets you see the types in more places.

Do you have the same issue in your IDE?

Reply:
Oh yeah, thanks for checking! Same for me (also vs code) and the expanded list is definitely better. I've changed it in https://github.com/vega/altair/pull/3515/commits/6fefd1270d59344bc2b44d7ee4491b8cb84b56ee